Cython/Compiler/TypeStubGenerator.py

```diff
@@ -0,0 +1,485 @@
+
+        if pos != -1:
+            parent_module = module_name[:pos]
+        
+        imported_name = node.lhs.name
+
+        if parent_module == imported_name:
+            self.visitchildren(node)
+            return node
+
+        self.putline("import %s as %s" % (module_name, imported_name))
+        return node 
+    
+    # Optimized orginal code by having there be one function to take 
+    # the place of two of them I could see what Scoder meant when 
+    # said the orginal pull request needed to be cleaned up...
+
+    def write_class(self, node, class_name):
+        print(node.__dict__)
+        self.put("class %s" % class_name)
+        if getattr(node,"bases",None) and isinstance(node.bases, TupleNode):
+            self.put("(")
+            self.put(",".join([name.name for name in node.bases.args]))
+            self.endline("):")
+        else:
+            self.put(":")
+
+        self.class_func_count = 0
+        self._visitchildren_indented(node)
+        if self.class_func_count < 1:
+            self.indent()
+            self.putline("pass")
+            self.dedent()
+        self.class_func_count = 0
+
+        self.emptyline()
+        return node 
+    
+    # I have tried to merege these before via visit_ClassDefNode but it causes the system to break so this 
+    # was the best I could do to minigate the problem - Vizonex 
+    def visit_CClassDefNode(self, node: CClassDefNode):
+        return self.write_class(node, node.class_name)
+
+    def visit_PyClassDefNode(self, node:PyClassDefNode):
+        return self.write_class(node, node.name)
+
+    def visit_CFuncDefNode(self, node: CFuncDefNode):
+        # cdefs are for C only...
+        if not node.overridable:
+            return node 
+
+        self.class_func_count += 1
+
+        func_name = node.declared_name()
+
+        self.startline()
+        self.put("def %s(" % func_name)
+        # Cleaned up alot of what the orginal author did by making a new function
+        self.put(", ".join(translate_annotations(node)))
+        # TODO Maybe Try passing docstrings in the future for vscode users' sake
+        # or have it also be a compiler argument?...
+        self.endline(") -> %s: ..." % ctype_name(node.type.return_type))
+        return node
+
+    def print_Decorator(self, decorator):
+        if isinstance(decorator, CallNode):
+            return
+        
+        self.startline("@")
+        if isinstance(decorator, NameNode):
+            self.endline("%s" % decorator.name)
+        else:
+            assert isinstance(decorator, AttributeNode) , "Decorator was not an attribute node..."
+            self.endline("%s.%s" % (decorator.obj.name,decorator.attribute))
+        
+
+    def annotation_Str(self, annotation:ExprNode) -> str:
+        return annotation.name if hasattr(annotation,"name") and annotation.is_name else  annotation.string.unicode_value 
+        
+     
+
+    def visit_DefNode(self,node:DefNode):
```

Comment:
Further to my comment about not supporting annotations, I'm also not keen on `node : DefNode` - it's absolutely explicit from the function already so it isn't adding useful information.

Right now it's irrelevant (since we can't have it) but it's definitely the type of annotation I'd be keen to avoid in future

Reply:
I'll be sure to fix all of this then. Thank you for your input